test/cluster/test_tablets.py

```diff
@@ -823,3 +823,135 @@ async def test_drop_keyspace_while_split(manager: ManagerClient):
     # release drop and wait for it to complete
     await manager.api.message_injection(servers[0].ip_addr, "truncate_compaction_disabled_wait")
     await drop_ks_task
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_tablet_split_finalization(manager: ManagerClient):
+    """
+    Reproducer for https://github.com/scylladb/scylladb/issues/21762
+        1) Start a node with config --target-tablet-size-in-bytes set to a low value to trigger split earlier
+        2) Inject error to block topology transition into split finalization handling
+        3) Create and populate `test` table
+        4) Wait for the table `test` to reach split finalization stage
+        5) Add a new server, to trigger tablet migration
+        6) Inject another error to force transition into tablet migration handling state
+        7) Expect finalization to be preferred over migration in `test` table
+    """
+    logger.info("Starting Node 1")
+    cfg = {'enable_user_defined_functions': False, 'enable_tablets': True}
+    cmdline = [
+        '--target-tablet-size-in-bytes=10240',
```

Comment:
In the future, prefer to use per-table tablet count hints to trigger splits (via `ALTER TABLE .. WITH tablets = {'min_tablet_count': xxx}` cql), it's more reliable, and faster (no stats propagation) than triggering by size.

Reply:
You didn't set `short_tablet_stats_refresh_interval`, isn't this test waiting for 1 minute before it starts splits? Which reinforces the above comment.

Reply:
Thank you! I have updated the code. The splits are more consistent and faster now.